lib/cylc/task_pool.py

```diff
@@ -756,6 +756,20 @@ def has_unkillable_tasks_only(self):
                     return False
         return True
 
+    def pool_is_stalled(self):
+        """Return True if no active, queued or clock trigger awaiting tasks"""
+        for itask in self.get_tasks():
+            if (itask.state.status in [TASK_STATUS_QUEUED,
+                                       TASK_STATUS_READY] or
+               itask.state.status in TASK_STATUSES_ACTIVE):
```

Comment:
The indentation *is* funny in order to pass the pep8 test - complains if the row doesn't start exactly there...